connectors/sources/notion.py

```diff
@@ -96,6 +96,48 @@ async def get_via_session(self, url):
             else:
                 raise
 
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
+    )
+    async def fetch_results(
+        self, function: Callable[..., Awaitable[Any]], next_cursor=None, **kwargs: Any
```

Comment:
why is `next_cursor` an explicit arg, instead of being included in `kwargs`? Below, it looks like this starts as 
```
kwargs.pop("start_cursor", None)
```
so I don't see the point.

Reply:
the `next_cursor` passed to `fetch_results` method is only equal to `start_cursor` during 1st iteration, hence that can be fetched via `kwargs.pop("start_cursor", None)`. 

But in the subsequent iterations of while loop, the `next_cursor` is available in the response of previous call. Hence, we cannot rely on `kwargs` for these iterations. Therefore we need to pass that explicitly along with `kwargs`